bpf/flows.c

```diff
@@ -171,7 +176,47 @@ static inline int flow_monitor(struct __sk_buff *skb, u8 direction) {
     }
     id.if_index = skb->ifindex;
     id.direction = direction;
+    u16 flags_t = 0;
+    if ((void *)eth + sizeof(struct ethhdr) <= data_end) {
+        if (bpf_ntohs(eth->h_proto) == ETH_P_IP) {
+            const struct iphdr *iph = (struct iphdr *)(data + sizeof(struct ethhdr));
+            if ((void *)iph + sizeof(struct iphdr) <= data_end) {
+                if (iph->protocol == IPPROTO_TCP) {
+                    const struct tcphdr *th = (struct tcphdr *)(data + sizeof(struct iphdr));
+                    if ((void *)th + sizeof(struct tcphdr) <= data_end) {
+                        //bpf_printk("fin=%d, syn=%d, rst=%d, psh=%d, ack=%d, urg=%d, ece=%d, cwr=%d, res1=%d doff=%d\n",th->fin, th->syn, th->rst, th->psh, th->ack,th->urg,th->ece, th->cwr, th->res1, th->doff);
+                        if (th->fin) {
+                            flags_t = setBit(flags_t, 0);
```

Comment:
for speed, I'd just remove the `setBit` function and do the following:

```c
#define FIN_FLAG 0b1
#define SYN_FLAG 0b10
#define RST_FLAG 0b100
```

Then replace each invocation to `setBit` by:
```c
if (th->fin) {
   flags_t |= FIN_FLAG
}
if (th->syn) {
    flags_t |= SYN_FLAG
}
if (th->rst) {
    flags_t |= RST_FLAG
}
etc...
```